docs/ref/migration-operations.txt

```diff
@@ -253,6 +253,11 @@ partial indexes.
 the database. On most database backends (all but PostgreSQL), Django will
 split the SQL into individual statements prior to executing them.
 
+.. warning::
+
+    Only use ``BEGIN`` or ``COMMIT`` in your SQL in :ref:`non-atomic migrations
+    <non-atomic-migrations>`, to avoid breaking Django's transaction state.
```

Comment:
As far as I'm concerned that's not true on Oracle. I'm not sure how it looks like on MySQL :thinking:

Reply:
Yeah it's a bit ambiguous.. migrations on MySQL are *all* non-atomic, even when you set `atomic = True`. I guess that's what you mean on Oracle too. That's explained in the linked section but maybe we could reword here better?

Reply:
"Non-atomic migrations" is about DDL transactions, but we can use `RunSQL()` for e.g. data migrations with DML operations. In such cases `BEGIN` also doesn't start atomic transaction on Oracle.

Using `COMMIT` in SQL statements behaves differently on each database and you need to know what you're doing, I don't think we should describe this in Django docs.

I agree that is can be surprising that `BEGIN` starts a transaction on SQLite and PostgreSQL, so maybe:
```
On PostgreSQL and SQLite, only use ``BEGIN`` in your SQL in ... 
```

What do you think?